tests/simpleDemo.py

```diff
@@ -45,6 +47,49 @@ def buildMITBatteryPaperPlan(datasetId):
 
     return mitPapers
 
+class VLDBPaperListing(pz.Schema):
+    """VLDBPaperListing represents a single paper from the VLDB conference"""
+    title = pz.Field(desc="The title of the paper", required=True)
+    authors = pz.Field(desc="The authors of the paper", required=True)
+    pdfLink = pz.Field(desc="The link to the PDF of the paper", required=True)
+
+def downloadVLDBPapers(vldbListingPageURLsId, outputDir):
+    """ This function downloads a bunch of VLDB papers from an online listing and saves them to disk.  It also saves a CSV file of the paper listings."""
+    policy = pz.MaxQuality()
+
+    # 1. Grab the input VLDB listing page(s) and scrape them for paper metadata
+    tfs = pz.Dataset(vldbListingPageURLsId, schema=pz.TextFile, desc="A file full of URLs of VLDB journal pages")
+    urls = tfs.convert(pz.URL, desc="The actual URLs of the VLDB pages", cardinality="oneToMany")  # oneToMany=True would be nice here.   
+    htmlContent = urls.map(pz.DownloadHTMLFunction())
+    vldbPaperListings = htmlContent.convert(VLDBPaperListing, desc="The actual listings for each VLDB paper", cardinality="oneToMany")
```

Comment:
While I don't think we should change it now, I think asking the user to provide a `cardinality` is a bit odd (it seems like something we should handle for them, but I understand that we can't know the `cardinality` a priori).

One half-baked idea for an alternative may be to introduce a `.flatten()` syntax, and if we see, e.g.:
```python
ds = pz.Dataset(...)
ds_new = ds.convert(...)
ds_newer = ds_new.flatten()
```
Then behind the scenes we effectively convert `ds.convert(...).flatten()` into `ds.convert(..., cardinality="oneToMany")`.

Reply:
I mainly think that managing cardinality is a core part of the data model of the universe. Sometimes I am OK leaving it up to the system, but in many cases I have strong constraints I want the system to enforce.  (Like: exactly one title, and >=1 authors)

Though I do think it's a bit weird to have it in the query syntax, rather than the schema declarations, where we ask the user to put most of their data model decisions.

Reply:
I don't think the current method is perfect and am open to suggestions